cmd/update-rules/main_test.go

```diff
@@ -163,3 +163,40 @@ func TestUpdateRules(t *testing.T) {
 		})
 	}
 }
+
+func TestDeleteRules(t *testing.T) {
+	tests := []struct {
+		name      string
+		branch    string
+		goVersion string
+	}{
+		{
+			"release-1.20",
+			"release-1.20",
+			"1.17.1",
+		},
+		{
+			"release-1.19",
+			"release-1.19",
+			"1.17.1",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			rules, err := load(testdataRules)
+			if err != nil {
+				t.Errorf("error loading test rules file %v", err)
+			}
+			UpdateRules(rules, tt.branch, tt.goVersion, true)
+
+			for _, repoRule := range rules.Rules {
```

Comment:
One more check needs to be done to make sure that, if the branch to be deleted was a non-existent branch, there are no changes to the rules that has been loaded from `testdataRules`

Reply:
we have the error logged in the below loop for the same if it failed to delete the branch do you mean should we check if branch exist or not  instead before deleting?

Reply:
> we have the error logged in the below loop for the same if it failed to delete the branch 

this is correct.

> do you mean should we check if branch exist or not instead before deleting?

no. The test should check that, if a non existent branch is tried for deletion, it doesnt affect rest of the rules.

Reply:
Do you have and recommendations around this